### ISSUE TITLE
Still supported?

### ISSUE THREAD

**Issue by jeregrine** [azukiapp/elixir-rethinkdb#12]
Hey are you thinking about still supporting this? <No pressure just want to get a gauge of interest on continuing this project> :)

**Comment by jeregrine** [comment]
Well if anyone is interested I've begun the process of upgrading to the latest elixir and to use bitwalker/exprotobuf https://github.com/jeregrine/elixir-rethinkdb/tree/upgrade_library

**Comment by nurugger07** [comment]
I'd be interested in looking at that :)

**Comment by jeregrine** [comment]
@nurugger07 Yes!! Well that rethink driver is an entire version behind the offical api and its pretty janky.

I got it to build but I never got it to run. I'm thinking a full rewrite would be better, as the new api uses JSON instead of protocolbuffers. But weirdly uses protocol buffs for the spec.

**Comment by nurugger07** [comment]
@jeregrine that might be a good idea. Just started playing with it this afternoon. I'll check out your changes

**Comment by jeregrine** [comment]
@nurugger07 I pushed my latest changes to upgrade_library

**Comment by neildaemond** [comment]
I'd also love to see this driver updated so I can put it to good use. @jeregrine are your updates working for you? I'm getting protobuf errrors like [issue#11](https://github.com/azukiapp/elixir-rethinkdb/issues/11)

**Comment by jeregrine** [comment]
I was able to get it to build on my fork/branch linked. I couldn’t get it to run.




This needs a new driver fully since the new spec doesn’t even use protocol bufs.

**Comment by neildaemond** [comment]
I agree, seems this driver won't work without any major changes.

**Comment by mossplix** [comment]
Anyone interested in upgrading it to latest elixir and rethinkdb?

**Comment by jeregrine** [comment]
Needs a rewrite unfortunately.

**Comment by nurugger07** [comment]
I'm starting a project next week that will be using it so I'm definitely interested. Last I checked though it does need a rewrite

**Comment by nuxlli** [comment]
Hello everybody,

I'm sorry about the delayed answers to the questions about this project. I've been busy the last 2 years with the project http://azk.io (a tool for developers) and I haven't had time to pay proper attention to this project (and its derived: https://github.com/azukiapp/elixir-protobuf).

In addiction, two things have happened which have made the maintenance of this project even harder:
- Elixir's final version has been published and many things have changed, mainly regarding the system of Records vs Map that deeply changes rethinkdb and protobuf libs' implementation;

- Support to JSON in the Rethinkdb's API, which has made the communication with the database even easier, but in the other hand, it has made unnecessary the use of elixir-protobuf, which is an important part of the current implementation.

If some of you is interested in maintain this project (or perform a complete code refactoring), you're welcome to contribute and I'll do my best to help.

**Comment by hamiltop** [comment]
Here's my weekend stab:
https://github.com/hamiltop/exrethinkdb

It's very incomplete and differs from elixir-rethinkdb in a few key ways:

1.  It uses the JSON API
2.  It uses pipelining to do parallel queries

Merging the two projects might make sense, depending on interest.

**Comment by jeregrine** [comment]
Amazing!

**Comment by Linicks** [comment]
All,
  I'm also interested in using RethinkDB with Elixir.  I just attempted to use this driver (azukiapp/elixir-rethinkdb) in hopes that it wouldn't be so broken that with a little poking and prodding it would work.  After fighting with the driver for a couple of hours, I was getting ready to start a thread similar to this one.  It sounds as though a rewrite is inevitable, and I'm happy to see that @hamiltop has started this process.  I need an Elixir based driver for a small project that I'm working on,  and I'm willing to help in this effort.  Unfortunately, I'm fairly new to Elixir, and have used many database drivers, but have never written one.  In may opinion, the Exrethinkdb driver is probably the best starting point because it's already designed around RethinkDB 2.x.  That being said, I was wondering if @nuxlli, and the other commenter’s here would be willing to help @hamiltop  with the Exrethinkdb driver?

-- Nick

**Comment by hamiltop** [comment]
@Linicks The "driver" portion is mostly done. Most of the work now is supporting the full range of query syntax. I'd be happy to point you in the right direction. (It's mostly just reading through rethinkdbs docs to see the spec)

Come on over and open an issue for a feature you would like that's not there yet and I can point you in the right direction.

**Comment by Linicks** [comment]
@hamiltop, thanks for all your work on this so far!  I will try to help out when and where I can, and I hope others will join in as well.  Having your guidance on this will help get over the initial learning curve.  Surprisingly, I just saw a great tutorial using the Exrethinkdb driver ( http://manukall.de/2015/04/25/building-a-simple-chat-app-in-elixir-with-phoenix-and-rethinkdb/ ).  How awesome is that :)

Cheers!
-- Nick

**Comment by hamiltop** [comment]
@nuxlli My client over at https://github.com/hamiltop/exrethinkdb is moving towards some degree of stability. I recently got a proposal at https://github.com/hamiltop/exrethinkdb to name it `RethinkDB` (and in my mind I would want the repo named `elixir-rethinkdb`). Any chance this library is going to be revived? Would you be open to exrethinkdb taking over the namespaces of this project?

**Comment by nuxlli** [comment]
@hamiltop finally had time to take a look at the implementation of https://github.com/hamiltop/exrethinkdb, and seemed very good.

I have interest in resuming the work on Elixir and Rethinkdb, but unfortunately without a prevision for this. Either way I do it, it could be on your version of the drive, ;)

To me you can use the namespace `RethinkDB` and I can do more: add a deprecate message in README and link to it to your implementation. 

What do you think?

**Comment by hamiltop** [comment]
The link would be appreciated. Thanks!

**Comment by nuxlli** [comment]
Change made: https://github.com/azukiapp/elixir-rethinkdb/commit/4f3af19199107ff8cda691157c00704a8137c664

I'm who is grateful for you to keep the elixir of integration work with Rethinkdb, these are two powerful technologies. Will make it easier to resume the work in the future to add support to Rethinkdb to https://github.com/elixir-lang/ecto ;)
